src/components/SingleArticle.jsx

```diff
@@ -0,0 +1,47 @@
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { getArticleById } from "../../api";
+
+function SingleArticle() {
+  const [singleArticle, setSingleArticle] = useState({});
+	const [loading, setLoading] = useState(true)
+  const { article_id } = useParams();
+
+
+  useEffect(() => {
+    getArticleById(article_id).then((data) => {
+      setSingleArticle(data);
+			setLoading(false)
+    });
+  }, [article_id]);
+	
+	
+
+	if (loading) {
+		return <><h2>Loading...</h2></>
```

Comment:
This doesn't need to be in a fragment (`<></>`) as you would already be returning one element, the `h2` 🔍